internal/operands/common-templates/reconcile.go

```diff
@@ -0,0 +1,128 @@
+package common_templates
+
+import (
+	"fmt"
+	templatev1 "github.com/openshift/api/template/v1"
+	core "k8s.io/api/core/v1"
+	rbac "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"kubevirt.io/ssp-operator/internal/common"
+	"path/filepath"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sync"
+)
+
+var (
+	loadTemplatesOnce sync.Once
+	templatesBundle   []templatev1.Template
+)
+
+func AddWatchTypesToScheme(s *runtime.Scheme) error {
+	return templatev1.Install(s)
+}
+
+func WatchClusterTypes() []runtime.Object {
+	return []runtime.Object{
+		&rbac.ClusterRole{},
+		&rbac.ClusterRoleBinding{},
+		&core.Namespace{},
+		&templatev1.Template{},
+	}
+}
+
+func Reconcile(request *common.Request) error {
+	for _, f := range []func(*common.Request) error{
+		reconcileGoldenImagesNS,
+		reconcileViewRole,
+		reconcileViewRoleBinding,
+		reconcileEditRole,
+		reconcileTemplates,
+	} {
+		if err := f(request); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func Cleanup(request *common.Request) error {
+	for _, obj := range []controllerutil.Object{
+		newGoldenImagesNS(goldenImagesNSname),
+		newViewRole(goldenImagesNSname),
+		newViewRoleBinding(goldenImagesNSname),
+		newEditRole(goldenImagesNSname),
+		//templatesBundle,
```

Comment:
If we currently don't want to remove the templates, maybe add a TODO comment, so we don't forget about it later.

Reply:
Changed this to append all the templates from the bundle